python/cuxfilter/charts/core/non_aggregate/core_non_aggregate.py

```diff
@@ -271,23 +271,23 @@ def query_chart_by_indices(
         if len(new_indices) == 0:
             # case: all selected indices were reset
             # reset the chart
-            self.reload_chart(self.source, False)
+            final_query = query
         elif len(new_indices) == 1:
             final_query = active_chart.x + "==" + str(float(new_indices[0]))
             if len(query) > 0:
                 final_query += " and " + query
-            # just a single index
-            self.reload_chart(
-                self.source.query(final_query), False,
-            )
         else:
             new_indices_str = ",".join(map(str, new_indices))
             final_query = active_chart.x + " in (" + new_indices_str + ")"
             if len(query) > 0:
                 final_query += " and " + query
-            self.reload_chart(
-                self.source.query(final_query), False,
-            )
+
+        self.reload_chart(
+            self.source.query(final_query)
+            if len(final_query) > 0
```

Comment:
Would there be downsides to letting `source.query` accept a zero length query?

Reply:
That results in a QuerySyntaxError
![image](https://user-images.githubusercontent.com/20476096/93628504-84b62c80-f99b-11ea-9b6f-80710829a0a6.png)

Reply:
Ah, I didn't realize `self.source` was not some wrapper we created. :+1: